File: src/main/java/net/liftweb/seventhings/snippet/MyWizard.java
package net.liftweb.seventhings.snippet;

import net.liftweb.http.AbstractScreen;
import net.liftweb.http.SJ;
import net.liftweb.http.js.JsCmd;
import net.liftweb.http.js.jquery.JqJsCmds;
import net.liftweb.seventhings.lib.JqJsCmdsJ;
import net.liftweb.seventhings.lib.ListJ;
import net.liftweb.seventhings.lib.WizardJ;
import net.liftweb.util.Func;
import net.liftweb.util.Func0;
import net.liftweb.wizard.Wizard;

/**
 * An example of a wizard in Lift
 */
public class MyWizard extends WizardJ {

    private MyWizardScala wizardScala = new MyWizardScala();

    private final MyWizardScala.ParentNameScreen parentName;
    private final MyWizardScala.FavoritePetScreen favoritePet;
    private final MyWizardScala.NameAndAgeScreen nameAndAge;

    public MyWizard() {
        parentName  = wizardScala.getParentName();
        favoritePet = wizardScala.getFavoritePet();
        nameAndAge  = wizardScala.getNameAndAge();
    }

    public JsCmd calcAjaxOnDone() {
        return JqJsCmdsJ.j().unblock();
    }

  // define the first screen

  // We ask the parent's name if the person is under 18
//  private WizardJ.ScreenJ parentNameObj = new WizardJ.ScreenJ() {
//    String parentName = fieldj(SJ.j().$qmark("Mom or Dad's name"), ListJ.toSeq(
//            valMinLen(Func.lift(new Func0<Integer>() {
//                public Integer apply() {
//                    return 2;
//                }
//            }), Func.lift(new Func0<String>() {
//                public String apply() {
//                    return SJ.j().$qmark("Name Too Short");
//                }
//            })),
//            valMaxLen(Func.lift(new Func0<Integer>() {
//                public Integer apply() {
//                    return 40;
//                }
//            }), Func.lift(new Func0<String>() {
//                public String apply() {
//                    return SJ.j().$qmark("Name Too Long");
//                }
//            }))));
//  };


  // we ask for the favorite pet
//  private WizardJ.Screen favoritePet = new WizardJ.Screen() {
//    String petName = AbstractScreen.field(SJ.j().$qmark("Pet's name", "",
//            AbstractScreen.valMinLen(2, SJ.j().$qmark("Name Too Short")),
//            AbstractScreen.valMaxLen(40, SJ.j().$qmark("Name Too Long"))));
//  };




//  // what to do on completion of the wizard
//  public void finish() {
//    SJ.j().notice("Thank you for registering your pet: " +
//            favoritePet.petName() +
//            " your age * 3: " + nameAndAge.age() * 3);
//  }

    public void finish() {
        wizardScala.finish();
    }

}

File: src/main/scala/net/liftweb/seventhings/snippet/MyWizard.scala
package net.liftweb.seventhings
package snippet

import net.liftweb._
import http._
import js.jquery.JqJsCmds._
import wizard._
import util.Helpers._

/**
 * An example of a wizard in Lift
 */
object MyWizard extends Wizard {
  override protected def calcAjaxOnDone = Unblock

  // define the first screen
  val nameAndAge = new Screen {
    // it has a name field
    val name = field(S ? "First Name", "",
                     valMinLen(2, S ? "Name Too Short"),
                     valMaxLen(40, S ? "Name Too Long"))

    // and an age field
    val age = field(S ? "Age", 0, minVal(5, S ? "Too young"),
      maxVal(120, S ? "You should be dead"))

    // choose the next screen based on the age
    override def nextScreen = if (age.is < 18) parentName else favoritePet
  }

  // We ask the parent's name if the person is under 18
  val parentName = new Screen {
    val parentName = field(S ? "Mom or Dad's name", "",
                           valMinLen(2, S ? "Name Too Short"),
      valMaxLen(40, S ? "Name Too Long"))
  }

  // we ask for the favorite pet
  val favoritePet = new Screen {
    val petName = field(S ? "Pet's name", "",
                        valMinLen(2, S ? "Name Too Short"),
                        valMaxLen(40, S ? "Name Too Long"))
  }

  // what to do on completion of the wizard
  def finish() {
    S.notice(S ? "Thank you for registering your pet" + ": "+
             favoritePet.petName+
             " " + S ? "your age" + "* 3: "+nameAndAge.age * 3)
  }
}

object AjaxRunner {
  def render = "* [onclick]" #> SHtml.ajaxInvoke(() => ModalDialog(<div><lift:MyWizard ajax="true"/></div>))
}
